### ISSUE TITLE
[wasm][debugger] `DebuggerTests.AssignmentTests.InspectVariableBeforeAndAfterAssignment` failing

### ISSUE THREAD

**Issue by radical** [dotnet/runtime#64188]
Failing on rolling build - [Build](https://dev.azure.com/dnceng/public/_build/results?buildId=1567770&view=logs&j=2cb5cbbb-10b2-57f5-cb06-88ab02df11cd&t=8e3b5183-1166-552f-0d9e-c49b4956b6ef), and [Log](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-main-bdd94cbc517447d787/DebuggerTests.AssignmentTests/1/console.649c87ae.log?sv=2019-07-07&se=2022-02-12T21%3A30%3A57Z&sr=c&sp=rl&sig=XZaiyDIp8TImXIx6TUJrAQ85XnuA3ygCSuDf71sMt9s%3D)

```
[21:35:25] info: Microsoft.WebAssembly.Diagnostics.TestHarnessProxy[0] New test request
[21:35:25] info: Microsoft.WebAssembly.Diagnostics.TestHarnessProxy[0] launching proxy for ws://127.0.0.1:37767/devtools/page/79E663E76834EA440CBAB0CD2CAB2BDB
[21:35:25] info: DebuggerTests.Inspector[0] waiting for the runtime to be ready
[21:35:27] info: DebuggerTests.Inspector[0] runtime ready, TEST TIME
[21:35:27] info: DebuggerTests.Inspector[0] console.log: ready
[21:35:27] fail: Microsoft.WebAssembly.Diagnostics.DevToolsProxy[0] sending error response for id: msg-:::9 -> [Result: IsOk: False, IsErr: True, Value: , Error: {   "message": "Unable to RuntimeGetProperties 'dotnet:scope:1'" } ]
[21:35:27] info: DebuggerTests.Inspector[0] console.trace: * Assertion at /__w/1/s/src/mono/mono/component/debugger-agent.c:2195, condition `<disabled>' not met 
[21:35:27] info: DebuggerTests.Inspector[0] console.error: * Assertion at /__w/1/s/src/mono/mono/component/debugger-agent.c:2195, condition `<disabled>' not met 
[21:35:27] dbug: Microsoft.WebAssembly.Diagnostics.InspectorClient[0] RunLoop stopped, reason: Shutdown. (parentToken: True, linked: True): 
[21:35:27] dbug: Microsoft.WebAssembly.Diagnostics.InspectorClient[0] Failing 0 pending cmds
[21:35:27] dbug: Microsoft.WebAssembly.Diagnostics.InspectorClient[0] Loop ended with socket: Aborted
[21:35:27] info: Microsoft.WebAssembly.Diagnostics.TestHarnessProxy[0] Proxy done
[xUnit.net 00:00:15.58]     DebuggerTests.AssignmentTests.InspectVariableBeforeAndAfterAssignment(clazz: "MONO_TYPE_ENUM", checkDefault: [[[...]], [[...]], [[...]], [[...]]], checkValue: [[[...]], [[...]], [[...]], [[...]]]) [FAIL]
[21:35:28] dbug: Microsoft.WebAssembly.Diagnostics.InspectorClient[0] connecting to ws://localhost:9400/launch-chrome-and-connect
[21:35:28] info: Microsoft.WebAssembly.Diagnostics.TestHarnessProxy[0] New test request
[21:35:28] info: Microsoft.WebAssembly.Diagnostics.TestHarnessProxy[0] launching proxy for ws://127.0.0.1:36459/devtools/page/1075ADF29D9088A2E8B9BF6E78493EBB
[21:35:28] info: DebuggerTests.Inspector[0] waiting for the runtime to be ready
  Failed DebuggerTests.AssignmentTests.InspectVariableBeforeAndAfterAssignment(clazz: "MONO_TYPE_ENUM", checkDefault: [[[...]], [[...]], [[...]], [[...]]], checkValue: [[[...]], [[...]], [[...]], [[...]]]) [153 ms]
  Error Message:
   [Runtime.getProperties returned False instead of True, for {
  "objectId": "dotnet:scope:1"
}, with Result: [Result: IsOk: False, IsErr: True, Value: , Error: {
  "message": "Unable to RuntimeGetProperties 'dotnet:scope:1'"
} ]]

Expected: True
Actual:   False
  Stack Trace:
     at DebuggerTests.DebuggerTestBase.AssertEqual(Object expected, Object actual, String label) in /_/src/mono/wasm/debugger/DebuggerTestSuite/DebuggerTestBase.cs:line 1058
   at DebuggerTests.DebuggerTestBase.GetProperties(String id, JToken fn_args, Nullable`1 own_properties, Nullable`1 accessors_only, Boolean expect_ok) in /_/src/mono/wasm/debugger/DebuggerTestSuite/DebuggerTestBase.cs:line 835
   at DebuggerTests.DebuggerTestBase.SendCommandAndCheck(JObject args, String method, String script_loc, Int32 line, Int32 column, String function_name, Func`2 wait_for_event_fn, Func`2 locals_fn, String waitForEvent) in /_/src/mono/wasm/debugger/DebuggerTestSuite/DebuggerTestBase.cs:line 512
   at DebuggerTests.DebuggerTestBase.StepAndCheck(StepKind kind, String script_loc, Int32 line, Int32 column, String function_name, Func`2 wait_for_event_fn, Func`2 locals_fn, Int32 times) in /_/src/mono/wasm/debugger/DebuggerTestSuite/DebuggerTestBase.cs:line 472
   at DebuggerTests.AssignmentTests.InspectVariableBeforeAndAfterAssignment(String clazz, JObject checkDefault, JObject checkValue) in /_/src/mono/wasm/debugger/DebuggerTestSuite/AssignmentTests.cs:line 48
--- End of stack trace from previous location ---
```

The runtime is asserting at https://github.com/dotnet/runtime/blob/main/src/mono/mono/component/debugger-agent.c#L2195:
```
...
[21:35:27] info: DebuggerTests.Inspector[0] console.trace: * Assertion at /__w/1/s/src/mono/mono/component/debugger-agent.c:2195, condition `<disabled>' not met 
[21:35:27] info: DebuggerTests.Inspector[0] console.error: * Assertion at /__w/1/s/src/mono/mono/component/debugger-agent.c:2195, condition `<disabled>' not met 
...
```

This was hit on the rolling build for https://github.com/dotnet/runtime/commit/4a6d169f5391532340074b0be4cab737c6ccdfc8 but was working on the previous build for https://github.com/dotnet/runtime/commit/5605bdff4b104496941cc9e6acfc19783a52e7b8 , and the new commit being: https://github.com/dotnet/runtime/commit/4a6d169f5391532340074b0be4cab737c6ccdfc8

```
Add runtime support for ref fields (#63985)
* Add mono and coreclr runtime support for ref fields

* Update Reflection.Emit tests to validate ref fields.

Add test for TypedReference as a ref field.
```

cc @AaronRobinsonMSFT 

```json
{
    "ErrorMessage": "Failed DebuggerTests.AssignmentTests.InspectVariableBeforeAndAfterAssignment",
    "BuildRetry": false
}
```

### Report

|Build|Definition|Test|Pull Request|
|---|---|---|---|
|[291985](https://dev.azure.com/dnceng-public/public/_build/results?buildId=291985)|dotnet/runtime|[chrome-DebuggerTests.AssignmentTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=291985&view=ms.vss-test-web.build-test-results-tab&runId=5851956&resultId=100275)|dotnet/runtime#84191|
|[291509](https://dev.azure.com/dnceng-public/public/_build/results?buildId=291509)|dotnet/runtime|[chrome-DebuggerTests.AssignmentTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=291509&view=ms.vss-test-web.build-test-results-tab&runId=5839670&resultId=100297)|dotnet/runtime#86722|
|[291438](https://dev.azure.com/dnceng-public/public/_build/results?buildId=291438)|dotnet/runtime|[chrome-DebuggerTests.AssignmentTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=291438&view=ms.vss-test-web.build-test-results-tab&runId=5838548&resultId=100280)|dotnet/runtime#86813|
|[291252](https://dev.azure.com/dnceng-public/public/_build/results?buildId=291252)|dotnet/runtime|[chrome-DebuggerTests.AssignmentTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=291252&view=ms.vss-test-web.build-test-results-tab&runId=5835228&resultId=100287)|dotnet/runtime#86501|
|[290651](https://dev.azure.com/dnceng-public/public/_build/results?buildId=290651)|dotnet/runtime|[chrome-DebuggerTests.AssignmentTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=290651&view=ms.vss-test-web.build-test-results-tab&runId=5817970&resultId=100293)|dotnet/runtime#84512|
|[290344](https://dev.azure.com/dnceng-public/public/_build/results?buildId=290344)|dotnet/runtime|[chrome-DebuggerTests.AssignmentTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=290344&view=ms.vss-test-web.build-test-results-tab&runId=5811180&resultId=100296)|dotnet/runtime#86925|
|[286567](https://dev.azure.com/dnceng-public/public/_build/results?buildId=286567)|dotnet/runtime|[chrome-DebuggerTests.AssignmentTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=286567&view=ms.vss-test-web.build-test-results-tab&runId=5710914&resultId=100281)|dotnet/runtime#84191|
|[286566](https://dev.azure.com/dnceng-public/public/_build/results?buildId=286566)|dotnet/runtime|[chrome-DebuggerTests.AssignmentTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=286566&view=ms.vss-test-web.build-test-results-tab&runId=5710902&resultId=100281)|dotnet/runtime#84191|
|[286306](https://dev.azure.com/dnceng-public/public/_build/results?buildId=286306)|dotnet/runtime|[chrome-DebuggerTests.AssignmentTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=286306&view=ms.vss-test-web.build-test-results-tab&runId=5704522&resultId=100280)|dotnet/runtime#86664|
|[285659](https://dev.azure.com/dnceng-public/public/_build/results?buildId=285659)|dotnet/runtime|[chrome-DebuggerTests.AssignmentTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=285659&view=ms.vss-test-web.build-test-results-tab&runId=5702964&resultId=100282)|dotnet/runtime#84191|
|[285663](https://dev.azure.com/dnceng-public/public/_build/results?buildId=285663)|dotnet/runtime|[chrome-DebuggerTests.AssignmentTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=285663&view=ms.vss-test-web.build-test-results-tab&runId=5689194&resultId=100282)|dotnet/runtime#84191|
|[284376](https://dev.azure.com/dnceng-public/public/_build/results?buildId=284376)|dotnet/runtime|[chrome-DebuggerTests.AssignmentTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=284376&view=ms.vss-test-web.build-test-results-tab&runId=5659886&resultId=100280)|dotnet/runtime#84191|
|[283355](https://dev.azure.com/dnceng-public/public/_build/results?buildId=283355)|dotnet/runtime|[chrome-DebuggerTests.AssignmentTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=283355&view=ms.vss-test-web.build-test-results-tab&runId=5640202&resultId=100166)|dotnet/runtime#86278|
|[280613](https://dev.azure.com/dnceng-public/public/_build/results?buildId=280613)|dotnet/runtime|[chrome-DebuggerTests.AssignmentTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=280613&view=ms.vss-test-web.build-test-results-tab&runId=5563646&resultId=100280)|dotnet/runtime#84191|
|[278935](https://dev.azure.com/dnceng-public/public/_build/results?buildId=278935)|dotnet/runtime|[chrome-DebuggerTests.AssignmentTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=278935&view=ms.vss-test-web.build-test-results-tab&runId=5532682&resultId=100280)|dotnet/runtime#85558|
|[276439](https://dev.azure.com/dnceng-public/public/_build/results?buildId=276439)|dotnet/runtime|[chrome-DebuggerTests.AssignmentTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=276439&view=ms.vss-test-web.build-test-results-tab&runId=5476964&resultId=100280)|dotnet/runtime#84191|
|[275141](https://dev.azure.com/dnceng-public/public/_build/results?buildId=275141)|dotnet/runtime|[chrome-DebuggerTests.AssignmentTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=275141&view=ms.vss-test-web.build-test-results-tab&runId=5444270&resultId=100280)|dotnet/runtime#86255|
|[274058](https://dev.azure.com/dnceng-public/public/_build/results?buildId=274058)|dotnet/runtime|[chrome-DebuggerTests.AssignmentTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=274058&view=ms.vss-test-web.build-test-results-tab&runId=5419650&resultId=100280)|dotnet/runtime#86255|
|[271904](https://dev.azure.com/dnceng-public/public/_build/results?buildId=271904)|dotnet/runtime|[chrome-DebuggerTests.AssignmentTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=271904&view=ms.vss-test-web.build-test-results-tab&runId=5370350&resultId=100280)|dotnet/runtime#86152|
|[267595](https://dev.azure.com/dnceng-public/public/_build/results?buildId=267595)|dotnet/runtime|[chrome-DebuggerTests.AssignmentTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=267595&view=ms.vss-test-web.build-test-results-tab&runId=5272110&resultId=100280)|dotnet/runtime#85730|
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|1|6|20|

**Comment by AaronRobinsonMSFT** [comment]
@lambdageek or @fanyang-mono I'm going to need a little help here. I've no idea why the `ref` field changes would have impacted a test related to debugger stepping around enum assignment. The test that is failing is

https://github.com/dotnet/runtime/blob/d964b638f1ad47a729ab2e6cf4d6822f76ea3e4f/src/mono/wasm/debugger/DebuggerTestSuite/AssignmentTests.cs#L42

The code being executed under that test is

https://github.com/dotnet/runtime/blob/d964b638f1ad47a729ab2e6cf4d6822f76ea3e4f/src/mono/wasm/debugger/tests/debugger-test/debugger-assignment-test.cs#L158-L165

**Comment by radical** [comment]
Cc @thaystg

**Comment by thaystg** [comment]
I'm working on it.

**Comment by thaystg** [comment]
@AaronRobinsonMSFT it's possibly unrelated with that commit. Still working on it.

**Comment by radical** [comment]
I didn't see the failure on the subsequent rolling build, so removing `blocking-clean-ci` label.

**Comment by radical** [comment]
Seeing this on windows too - [log](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-main-742d7512525c4783bc/DebuggerTests.AssignmentTests/1/console.ec3ac5f5.log?sv=2019-07-07&se=2022-02-18T09%3A53%3A44Z&sr=c&sp=rl&sig=T9yySQ6xDMkHNfap55lVjZIqJo2QE89IR3LS1tU4%2FT0%3D).

```
console.trace: * Assertion at D:/a/_work/1/s/src/mono/mono/component/debugger-agent.c:2195, condition `<disabled>' not met

console.error: * Assertion at D:/a/_work/1/s/src/mono/mono/component/debugger-agent.c:2195, condition `<disabled>' not met
```

**Comment by sbomer** [comment]
This is continuing to show up in rolling builds - see the associated builds in the original issue up top. Do we have confirmation that it's introduced by the ref field changes? Could we revert the changes that are causing these or temporarily disable the tests to unblock ci?

**Comment by robertmclaws** [comment]
Just an FYI, this issue is not limited to just the testing framework. I'm experiencing it at runtime in my Blazor apps.

**Comment by radical** [comment]
Can you please file a separate issue with the details?

**Comment by robertmclaws** [comment]
I can, I just wanted to point out that the problem is not in the test runners themselves but in .NET 6.0. IN case you were trying to trace down a test problem.

**Comment by radical** [comment]
I haven't seen this recently, but hit it today on https://github.com/dotnet/runtime/pull/63894 . [build](https://dev.azure.com/dnceng/public/_build/results?buildId=1628396&view=logs&j=2b5b5a32-5b33-52cd-8502-602252fb8ecd&t=ade3ad67-4440-59af-8c75-fa6108b65754):

```
[18:22:14] dbug: Microsoft.WebAssembly.Diagnostics.InspectorClient[0] connecting to ws://localhost:9400/launch-chrome-and-connect
[18:22:14] info: Microsoft.WebAssembly.Diagnostics.TestHarnessProxy[0] New test request
[18:22:15] info: Microsoft.WebAssembly.Diagnostics.TestHarnessProxy[0] launching proxy for ws://127.0.0.1:49729/devtools/page/7FC16A3E6124A3AE52D98214E18F7A98
[18:22:15] info: DebuggerTests.Inspector[0] waiting for the runtime to be ready
[18:22:15] info: DebuggerTests.Inspector[0] console.warning: Warning: Unable to read debug information of: debugger-test-with-full-debug-type.dll (use DebugType=Portable/Embedded)
[18:22:16] info: DebuggerTests.Inspector[0] console.log: ready
[18:22:16] info: DebuggerTests.Inspector[0] runtime ready, TEST TIME
[18:22:16] fail: Microsoft.WebAssembly.Diagnostics.DevToolsProxy[0] sending error response for id: msg-:::9 -> [Result: IsOk: False, IsErr: True, Value: , Error: {   "message": "Unable to RuntimeGetProperties 'dotnet:scope:1'" } ]
[18:22:16] dbug: Microsoft.WebAssembly.Diagnostics.InspectorClient[0] RunLoop stopped, reason: Shutdown. (parentToken: False, linked: False): 
[18:22:16] info: Microsoft.WebAssembly.Diagnostics.TestHarnessProxy[0] Proxy done
[18:22:16] dbug: Microsoft.WebAssembly.Diagnostics.InspectorClient[0] Failing 0 pending cmds
[18:22:16] dbug: Microsoft.WebAssembly.Diagnostics.InspectorClient[0] Loop ended with socket: CloseSent
[18:22:16] dbug: Microsoft.WebAssembly.Diagnostics.InspectorClient[0] connecting to ws://localhost:9400/launch-chrome-and-connect
[18:22:16] info: Microsoft.WebAssembly.Diagnostics.TestHarnessProxy[0] New test request
[xUnit.net 00:00:22.05]     DebuggerTests.AssignmentTests.InspectVariableBeforeAndAfterAssignment(clazz: "MONO_TYPE_VALUETYPE", checkDefault: [[[...]], [[...]], [[...]], [[...]]], checkValue: [[[...]], [[...]], [[...]], [[...]]]) [FAIL]
[18:22:16] info: Microsoft.WebAssembly.Diagnostics.TestHarnessProxy[0] launching proxy for ws://127.0.0.1:49810/devtools/page/7B9400DFDB63B32330514453EC0D9033
[18:22:16] info: DebuggerTests.Inspector[0] waiting for the runtime to be ready
  Failed DebuggerTests.AssignmentTests.InspectVariableBeforeAndAfterAssignment(clazz: "MONO_TYPE_VALUETYPE", checkDefault: [[[...]], [[...]], [[...]], [[...]]], checkValue: [[[...]], [[...]], [[...]], [[...]]]) [289 ms]
  Error Message:
   [Runtime.getProperties returned False instead of True, for {
  "objectId": "dotnet:scope:1"
}, with Result: [Result: IsOk: False, IsErr: True, Value: , Error: {
  "message": "Unable to RuntimeGetProperties 'dotnet:scope:1'"
} ]]

Expected: True
Actual:   False
  Stack Trace:
     at DebuggerTests.DebuggerTestBase.AssertEqual(Object expected, Object actual, String label) in /_/src/mono/wasm/debugger/DebuggerTestSuite/DebuggerTestBase.cs:line 1081
   at DebuggerTests.DebuggerTestBase.GetProperties(String id, JToken fn_args, Nullable`1 own_properties, Nullable`1 accessors_only, Boolean expect_ok) in /_/src/mono/wasm/debugger/DebuggerTestSuite/DebuggerTestBase.cs:line 858
   at DebuggerTests.DebuggerTestBase.SendCommandAndCheck(JObject args, String method, String script_loc, Int32 line, Int32 column, String function_name, Func`2 wait_for_event_fn, Func`2 locals_fn, String waitForEvent) in /_/src/mono/wasm/debugger/DebuggerTestSuite/DebuggerTestBase.cs:line 535
   at DebuggerTests.DebuggerTestBase.StepAndCheck(StepKind kind, String script_loc, Int32 line, Int32 column, String function_name, Func`2 wait_for_event_fn, Func`2 locals_fn, Int32 times) in /_/src/mono/wasm/debugger/DebuggerTestSuite/DebuggerTestBase.cs:line 472
   at DebuggerTests.AssignmentTests.InspectVariableBeforeAndAfterAssignment(String clazz, JObject checkDefault, JObject checkValue) in /_/src/mono/wasm/debugger/DebuggerTestSuite/AssignmentTests.cs:line 48
--- End of stack trace from previous location ---
```

This time
- it is for `MONO_TYPE_VALUETYPE`
- no assertions

**Comment by radical** [comment]
This is being [hit again](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-76605-merge-2e46162e5d0948aea9/chrome-DebuggerTests/1/console.e341a8f3.log?helixlogtype=result), but with a different error:

```
[xUnit.net 00:03:31.48]     DebuggerTests.AssignmentTests.InspectVariableBeforeAndAfterAssignment(clazz: "MONO_TYPE_ENUM", checkDefault: [[[...]], [[...]], [[...]], [[...]]], checkValue: [[[...]], [[...]], [[...]], [[...]]], methodName: "DebuggerTests.StepInTest<DebuggerTests.RGB>.Tested"...) [FAIL]
  Failed DebuggerTests.AssignmentTests.InspectVariableBeforeAndAfterAssignment(clazz: "MONO_TYPE_ENUM", checkDefault: [[[...]], [[...]], [[...]], [[...]]], checkValue: [[[...]], [[...]], [[...]], [[...]]], methodName: "DebuggerTests.StepInTest<DebuggerTests.RGB>.Tested"...) [84 ms]
  Error Message:
   Assert.Equal() Failure
Expected: 2
Actual:   1
  Stack Trace:
     at DebuggerTests.AssignmentTests.<>c__DisplayClass3_0.<<InspectVariableBeforeAndAfterAssignment>b__0>d.MoveNext() in /_/src/mono/wasm/debugger/DebuggerTestSuite/AssignmentTests.cs:line 58
--- End of stack trace from previous location ---
   at DebuggerTests.DebuggerTestBase.SendCommandAndCheck(JObject args, String method, String script_loc, Int32 line, Int32 column, String function_name, Func`2 wait_for_event_fn, Func`2 locals_fn, String waitForEvent) in /_/src/mono/wasm/debugger/DebuggerTestSuite/DebuggerTestBase.cs:line 621
   at DebuggerTests.DebuggerTestBase.StepAndCheck(StepKind kind, String script_loc, Int32 line, Int32 column, String function_name, Func`2 wait_for_event_fn, Func`2 locals_fn, Int32 times) in /_/src/mono/wasm/debugger/DebuggerTestSuite/DebuggerTestBase.cs:line 555
   at DebuggerTests.AssignmentTests.InspectVariableBeforeAndAfterAssignment(String clazz, JObject checkDefault, JObject checkValue, String methodName) in /_/src/mono/wasm/debugger/DebuggerTestSuite/AssignmentTests.cs:line 55
--- End of stack trace from previous location ---
[Result: IsOk: True, IsErr: False, Value: {
  "breakpointId": "1:539:8:dotnet://debugger-test.dll//_/src/mono/wasm/debugger/tests/debugger-test/debugger-test.cs",
  "locations": [
    {
      "scriptId": "dotnet://27312_1",
      "lineNumber": 542,
      "columnNumber": 8
    }
  ]
}, Error:  ]
[Result: IsOk: True, IsErr: False, Value: {
  "breakpointId": "1:1272:8:dotnet://debugger-test.dll//_/src/mono/wasm/debugger/tests/debugger-test/debugger-test.cs",
  "locations": []
}, Error:  ]
```

Also [here](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-76606-merge-49e8ce3c210042ee87/chrome-DebuggerTests/1/console.5a1daadc.log?helixlogtype=result).

**Comment by ulisesh** [comment]
The error message has to improve, build analysis will match this known issue with any failed build or test that contains the string `DebuggerTests.AssignmentTests.InspectVariableBeforeAndAfterAssignment`. For example, [this build](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/93017/logs/1981)

A better error message would be `Failed DebuggerTests.AssignmentTests.InspectVariableBeforeAndAfterAssignment`

**Comment by radical** [comment]
I have removed the `known build error` label because we don't have a unique string to match this error. The current one being used matches *any* failure in this test, and that can end up covering other unrelated genuine failures.

**Comment by lewing** [comment]
@thaystg what is the status on this one?

**Comment by thaystg** [comment]
I will close this one, me and @radical haven't seen this recently.
